src/arrayMethodSort.js

```diff
@@ -6,6 +6,27 @@
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
     // write code here
+    const compare = compareFunction || function(a, b) {
+      if (a < b) {
+        return -1;
+      }
+
+      if (a > b) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare((this[j]).toString(), this[j + 1].toString()) > 0) {
```

Comment:
You're converting elements to strings before comparison. This will lead to incorrect results if the array contains numbers or other non-string elements. You should pass the original elements to the compare function.